sdk/core/System.ClientModel/tests/Message/PipelineMessageTests.cs

```diff
@@ -111,6 +119,45 @@ public void TryGetTypeKeyedPropertyReturnsCorrectValues()
         Assert.AreEqual(4444, ((T4)value!).Value);
     }
 
+    [Test]
+    public async Task UnbufferedStreamAccessibleAfterMessageDisposed()
+    {
+        byte[] serverBytes = new byte[1000];
+        new Random().NextBytes(serverBytes);
+
+        ClientPipelineOptions options = new() { NetworkTimeout = Timeout.InfiniteTimeSpan };
+        ClientPipeline pipeline = ClientPipeline.Create(options);
+
+        using TestServer testServer = new(async context =>
+        {
+            await context.Response.Body.WriteAsync(serverBytes, 0, serverBytes.Length).ConfigureAwait(false);
+        });
+
+        PipelineResponse? response;
+        using (PipelineMessage message = pipeline.CreateMessage())
+        {
+            message.Request.Uri = testServer.Address;
+            message.BufferResponse = false;
+
+            await pipeline.SendSyncOrAsync(message, IsAsync);
+
+            response = message.ExtractResponse();
```

Comment:
do we have tests checking if this works when the response is buffered?

Reply:
Good suggestion

Reply:
I updated the one test to run with test cases for message.BufferResponse true and again with false.